### PR TITLE
improved detection of requested number of instances in scale-up action

### PR DESCRIPTION
**Pull request by konstan**
Connected to #548

### CODE REVIEW

File: jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java [hunk 1]
```diff
@@ -20,13 +20,12 @@
...
-import java.util.Arrays;
-import java.util.List;
-
-import javax.persistence.EntityManager;
-import javax.persistence.EntityTransaction;
-
+import com.sixsq.slipstream.configuration.Configuration;
+import com.sixsq.slipstream.exceptions.*;
+import com.sixsq.slipstream.factory.DeploymentFactory;
+import com.sixsq.slipstream.persistence.*;
```
Comment (loomis) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java]: Avoid wildcard imports.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java [hunk 2]
```diff
@@ -20,13 +20,12 @@
...
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
-import javax.persistence.EntityManager;
-import javax.persistence.EntityTransaction;
-
+import com.sixsq.slipstream.configuration.Configuration;
+import com.sixsq.slipstream.exceptions.*;
```
Comment (loomis) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java]: Avoid wildcard imports.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java [hunk 3]
```diff
@@ -257,11 +244,27 @@ private boolean isIgnoreAbort() {
...
-					NUMBER_INSTANCES_ADD_DEFAULT));
+			if (formParams.isEmpty()) {
+				return Integer.parseInt(NUMBER_INSTANCES_ADD_DEFAULT);
+			}
+			if (!formParams.contains(NUMBER_INSTANCES_ADD_FORM_PARAM)) {
+				throw new ResourceException(Status.CLIENT_ERROR_BAD_REQUEST,
+						"No form parameter " + NUMBER_INSTANCES_ADD_FORM_PARAM +
+								"=# found in the scale-up request.");
+			}
+			return Integer.parseInt(form.getFirstValue(NUMBER_INSTANCES_ADD_FORM_PARAM));
```
Comment (loomis) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java]: Wouldn't it be clearer to have the usual case first?
```
if (form.Params.contains(NUMBER_INSTANCES_ADD_FORM_PARAM)) {
 ...
} else if (formParams.isEmpty()) {
...
} else {
throw...
}
```

File: jar-service/src/test/java/com/sixsq/slipstream/run/RunNodeResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	@Test
+	public void getNumberOfInstancesToAdd() {
+		assert Integer.parseInt(RunNodeResource.NUMBER_INSTANCES_ADD_DEFAULT) ==
+				RunNodeResource.getNumberOfInstancesToAdd(new Form(""));
+		assert Integer.parseInt(RunNodeResource.NUMBER_INSTANCES_ADD_DEFAULT) ==
+				RunNodeResource.getNumberOfInstancesToAdd(new Form("n=1"));
+		assert 0 == RunNodeResource.getNumberOfInstancesToAdd(new Form("n=0"));
+		assert 0 == RunNodeResource.getNumberOfInstancesToAdd(new Form("n=0&foo=bar"));
+		assert 0 == RunNodeResource.getNumberOfInstancesToAdd(new Form("foo=bar&n=0"));
```
Comment (loomis) [inline jar-service/src/test/java/com/sixsq/slipstream/run/RunNodeResourceTest.java]: A bit strange to be using raw asserts as tests.  Shouldn't you be using `assertTrue` or `assertEquals` form the JUnit test framework?

Comment (konstan) [inline jar-service/src/test/java/com/sixsq/slipstream/run/RunNodeResourceTest.java]: Grr... Just forgot to substitute.

File: jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java (hunk 2 above)
Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/run/RunNodeResource.java]: Need to update configuration in IntelliJ.

**Comment by konstan** [comment]
@loomis all your comments were addressed.  Please check.

**Pull request merged**